artiq/frontend/aqctl_corelog.py

```diff
@@ -50,26 +58,29 @@ async def get_logs(host):
     writer.write(struct.pack("B", Request.PullLog.value))
     await writer.drain()
 
-    while True:
-        length, = struct.unpack(endian + "l", await reader.readexactly(4))
-        log = await reader.readexactly(length)
-
-        for line in log.decode("utf-8").splitlines():
-            m = re.match(r"^\[.+?\] (TRACE|DEBUG| INFO| WARN|ERROR)\((.+?)\): (.+)$", line)
-            levelname = m.group(1)
-            if levelname == 'TRACE':
-                level = logging.TRACE
-            elif levelname == 'DEBUG':
-                level = logging.DEBUG
-            elif levelname == ' INFO':
-                level = logging.INFO
-            elif levelname == ' WARN':
-                level = logging.WARN
-            elif levelname == 'ERROR':
-                level = logging.ERROR
-            name = 'firmware.' + m.group(2).replace('::', '.')
-            text = m.group(3)
-            log_with_name(name, level, text)
+    try:
+        while True:
+            length, = struct.unpack(endian + "l", await reader.readexactly(4))
+            log = await reader.readexactly(length)
+
+            for line in log.decode("utf-8").splitlines():
+                m = re.match(r"^\[.+?\] (TRACE|DEBUG| INFO| WARN|ERROR)\((.+?)\): (.+)$", line)
+                levelname = m.group(1)
+                if levelname == 'TRACE':
+                    level = logging.TRACE
+                elif levelname == 'DEBUG':
+                    level = logging.DEBUG
+                elif levelname == ' INFO':
+                    level = logging.INFO
+                elif levelname == ' WARN':
+                    level = logging.WARN
+                elif levelname == 'ERROR':
+                    level = logging.ERROR
+                name = 'firmware.' + m.group(2).replace('::', '.')
+                text = m.group(3)
+                log_with_name(name, level, text)
+    except TimeoutError:
+        log_with_name("core_log", logging.ERROR, "Terminated due to connection failure")
```

Comment:
You only addressed one part of my comment.
Also I suspect there are other exceptions than ``TimeoutError`` that can be raised when the core device disconnects.

Reply:
If you're putting **only** ``variable = await readexactly(something_trivial)`` into a ``try`` block then just ``except:`` might be the simplest solution.

Reply:
And still print the exc_info to

Reply:
at least at the debug level, see what commmoninj does and make consistent

Reply:
> Also I suspect there are other exceptions

I think `IncompleteReadError` might occur sometime, but it might occur without disconnecting core device. 
Maybe log all other types of error without termination?

Reply:
>  Maybe log all other types of error without termination?

Why would such errors be trivially recoverable and not warrant termination?